docs/source/format/Columnar.rst

```diff
@@ -765,6 +765,66 @@ application.
 We discuss dictionary encoding as it relates to serialization further
 below.
 
+.. _run-length-encoded-layout:
+
+Run-Length Encoded Layout
+-------------------------
+
+Run-Length is a data representation that represents data as sequences of the
+same value, called runs. Each run is represented as a value, and an integer
+describing how often this value is repeated.
+
+Any array can be run-length encoded. A run-length encoded array has no buffers
+by itself, but has two child arrays. The first one holds a signed 32-bit integer
+for each run. The actual values of each run are held the second child array.
+
+The values in the first child array represent the length of each run. They do
+not hold the length of the respective run directly, but the accumulated length
+of all runs from the first to the current one, i.e. the logical index where the
+current run ends. This allows relatively efficient random access from a logical
+index using binary search. The length of an individual run can be determined by
+subtracting two adjacent values.
+
+A run has to have a length of at least 1. This means the values in the
+run ends array all positive and in strictly ascending order. A run end cannot be
+null.
+
+As an example, you could have the following data: ::
+
+    type: Float32
+    [1.0, 1.0, 1.0, 1.0, null, null, 2.0]
+
+In Run-length-encoded form, this could appear as:
+
+::
+
+    * Length: 7, Null count: 2
+    * Children arrays:
+
+      * run ends (Int32):
+        * Length: 3, Null count: 0
+        * Validity bitmap buffer: Not required
```

Comment:
```suggestion
        * Validity bitmap buffer: Not present (not allowed)
```
See comment above

Reply:
I still think it would be best to be explicit here that no validity bitmap is allowed to avoid ambiguity